script/develop.sh

```diff
@@ -32,4 +34,4 @@ cleanup() {
 trap cleanup EXIT INT
 
 # start a watcher. This loops forever, so we don't need to loop ourselves.
-find src styleguide-app -type f -not -ipath '*elm-stuff*' | entr -c -p make public
+git ls-files | entr -c -p shake --compact "$SHAKE_TARGET"
```

Comment:
For some reason I'm getting `Shakefile.hs: getCurrentDirectory:getWorkingDirectory: resource exhausted (Too many open files)` when running shake with entr (and it's only with entr) and when setting `shakeThreads = 0,` or any number higher than 13 in my system.

With 14 even a Shakefile that does basically nothing fails with that error.

I'm gonna open an issue about it in the shake repo.

Reply:
What's your ulimit like? Maybe `git ls-files` has juuuuuuuust enough to bump you close to the limit. `entr` opens a ton of handles.

Reply:
```
$ ulimit
unlimited
$ ulimit -s
8192
$ ulimit -u
11136
```

Does it seem low? The repo is so small, `git ls-files` has 174 files.

Reply:
Oh, it's:

```
$ ulimit -n
256
```

Raising it to like 1k fixes it. Still, how do we manage to watch stuff in the monorepo without having to resort to this?

Reply:
I think that's the only way to do it. If `entr` fails with that it links to a help page.

Maybe `find` produced fewer files though? 🤷

Reply:
Oh, I'm going to guess it's bc we don't use `entr`, but `watchman`, from this review of watchers: https://anarc.at/blog/2019-11-20-file-monitoring-tools/#entr

Reply:
So 💭 either set a high `ulimit -n` in this file (1k?) or ditch `entr`?

Reply:
ok, switched to `watchexec` in 6236588. How's that work for you?